### PR TITLE
Modify all functions for templates and Insert new features

### PR DESCRIPTION
**Pull request by pacospace**
Depends-On: https://github.com/thoth-station/storages/pull/998

- Improved readability
- Standardized naming
- Improved functions generalization
- Added `save_results` flag
- Added violin plots
- Added `project_folder` flag

### CODE REVIEW

File: thoth/lab/inspection.py [hunk 1]
```diff
@@ -709,15 +745,13 @@ def plot_interpolated_statistics_of_inspection_parameters(
...
-    df_inspection_batches_dict: dict, inspection_identifiers: List[str], n_parallel: int = 6
-) -> pd.DataFrame():
+def create_inspection_time_dataframe(df_inspection_batches_dict: dict, n_parallel: int = 6) -> pd.DataFrame():
     """Create pd.Dataframe of time of inspections for build and job."""
     tot_time_builds = []
     tot_time_jobs = []
     tot_time_sum_builds_and_jobs = []
 
-    for identifier, dataframe in df_inspection_batches_dict.items():
+    for identifier, dataframe in list(df_inspection_batches_dict.items()):
```
Comment (CermakM) [inline thoth/lab/inspection.py]: is the `list(...)` necessary here?

Comment (pacospace) [reply thoth/lab/inspection.py]: I have to correct it, thanks Marek :beers:

**Comment by pacospace** [comment]
Updated :)

**Comment by pacospace** [comment]
Updated! All functions are ready

**Comment by CermakM** [comment]
This has become really hard to navigate with all those force pushes and is no longer an 'Update description and little changes' thing, can we finish the initial intention of the PR first before pushing new features to this branch?

**Pull request merged**